modules/es-extensions/publisher/modules/greg-publisher-api.js

```diff
@@ -468,5 +469,41 @@ var gregAPI = {};
             discoveryEnumData.orphanArtifactStrategy.push(OrphanArtifactStrategy.values()[index].name());
         }
         return discoveryEnumData;
+    };
+
+    gregAPI.password.addNewPassword = function (session, type, key, value) {
+        var am = assetManager(session, type);
+        var registry = am.registry.registry;
+        // Collection path used to store key and encrypted password value.
+        var path = "/_system/config/repository/components/secure-vault";
+        var resource;
+
+        if(registry.resourceExists(path)){
```

Comment:
Hi Prabath,
Thanks for pointing this out.
This has been fixed in PR #348 .
Thanks